api/auth/auth.py

```diff
@@ -0,0 +1,67 @@
+from api.models import User, db
+from flask import request, jsonify, g, Blueprint
+from flask_httpauth import HTTPTokenAuth
+from itsdangerous import TimedJSONWebSignatureSerializer as Serializer, SignatureExpired, BadSignature
+import json
+
+auth = HTTPTokenAuth(scheme='Bearer')
+
+auth_view = Blueprint('auth_view', __name__, url_prefix='/auth')
+
+
+@auth.verify_token
+def verify_auth_token(token):
+    if not token:
+        return jsonify({"message": "supply token"}), 401
+    user_id = User.verify_auth_token(token=token)
+    if not token:
+        return False
+    g.user = db.session.query(User).filter_by(id=user_id).first()
+    return True
+
+
+def verify_password(username, password):
+    user = db.session.query(User).filter_by(username=username).first()
+    if not user or not user.verify_password(password):
+        return False
+    return user
+
+
+@auth.error_handler
+def unauthorised_error():
```

Comment:
Consider creating another file for error handlers exclusively. Its key for separation of concerns.